simplesat/dependency_solver.py

```diff
@@ -52,12 +52,7 @@ def _create_rules(self, request):
             pool.package_id(package)
             for package in pool.what_provides(requirement)
         ]
-        self._policy.add_packages_by_id(requirement_ids)
-
-        # Add installed packages.
```

Comment:
from an API POV, I wonder if it would not be simpler to remove `add_packages_by_id` altogether, and make it an implementation detail (called at creation time from a passed `installed_repository`).